src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,40 @@
 package core.mate.academy.service;
 
-/**
- * Your implementation of MachineService.
- */
-public class MachineServiceImpl {
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            MachineProducer<? extends Machine> buldozerProducer = new BulldozerProducer();
+            return (List<Machine>) buldozerProducer.get();
+        } else if (Excavator.class.equals(type)) {
+            MachineProducer<? extends Machine> excavatorProducer = new ExcavatorProducer();
+            return (List<Machine>) excavatorProducer.get();
+        } else if (Truck.class.equals(type)) {
+            MachineProducer<? extends Machine> truckProducer = new TruckProducer();
+            return (List<Machine>) truckProducer.get();
+        }
```

Comment:
you can make those producers as global variables, then you wont create an instance of producer each time you invoke the method